ixmp/core.py

```diff
@@ -90,26 +81,25 @@ def __init__(self, dbprops=None, dbtype=None, jvmargs=None):
 
         try:
             # if no dbtype is specified, launch Platform with properties file
-            # or use 'config/default.properties' file
             if dbtype is None:
-                dbprops = dbprops or 'default.properties'
-                dbprops = dbprops if os.path.isfile(dbprops) \
-                    else os.path.join(ix.default_paths.CONFIG_DIR, dbprops)
-                print("launching ixmp.Platform using config file at '{}'"
-                      .format(dbprops))
+                dbprops = default_dbprops_file() if dbprops is None \
+                    else find_dbprops(dbprops)
+                logger().info("launching ixmp.Platform using config file at '{}'"
+                              .format(dbprops))
                 self._jobj = java.ixmp.Platform("Python", dbprops)
             # if dbtype is specified, launch Platform with local database
-            else:
-                dbprops = dbprops or os.path.join(local_path,
-                                                  'localdb', 'default')
-                print("launching ixmp.Platform with local {} database at '{}'"
-                      .format(dbtype, dbprops))
+            elif dbtype == 'HSQLDB':
```

Comment:
So I added this explicit check for HSQLDB. We can take it out, but I think it's worth it (esp if we get other backends). I do *not* understand how this plays with our tests, though... see comment below.